backend/src/nodes/image_chan_nodes.py

```diff
@@ -27,39 +27,91 @@ def __init__(self):
         )
         self.inputs = [ImageInput()]
         self.outputs = [
-            ImageOutput("Blue Channel"),
-            ImageOutput("Green Channel"),
-            ImageOutput("Red Channel"),
-            ImageOutput("Alpha Channel"),
+            ImageOutput("R Channel").with_id(2),
+            ImageOutput("G Channel").with_id(1),
+            ImageOutput("B Channel").with_id(0),
+            ImageOutput("A Channel"),
         ]
         self.category = IMAGE_CHANNEL
-        self.name = "Split Channels"
+        self.name = "Separate RGBA"
         self.icon = "MdCallSplit"
         self.sub = "All"
 
     def run(self, img: np.ndarray) -> list[np.ndarray]:
         """Split a multi-channel image into separate channels"""
 
-        c = 1
-        dtype_max = 1
-        try:
-            dtype_max = np.iinfo(img.dtype).max
-        except:
-            logger.debug("img dtype is not int")
+        h, w, c = get_h_w_c(img)
+        safe_out = np.ones((h, w))
 
-        if img.ndim > 2:
-            c = img.shape[2]
-            safe_out = np.ones_like(img[:, :, 0]) * dtype_max
-        else:
-            safe_out = np.ones_like(img) * dtype_max
+        if img.ndim == 2:
+            return [img, safe_out, safe_out, safe_out]
+
+        c = min(c, 4)
 
         out = []
         for i in range(c):
             out.append(img[:, :, i])
         for i in range(4 - c):
             out.append(safe_out)
 
-        return out
+        return [out[2], out[1], out[0], out[3]]
+
+
+@NodeFactory.register("chainner:image:combine_rgba")
+class CombineRgbaNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = (
+            "Merges the given channels together and returns an RGBA image."
+            " If RGB or RGBA image is given as a channel, it will be converted to grayscale first."
+        )
+        self.inputs = [
+            ImageInput("R Channel"),
+            ImageInput("G Channel"),
+            ImageInput("B Channel"),
+            ImageInput("A Channel").make_optional(),
+        ]
+        self.outputs = [ImageOutput()]
+        self.category = IMAGE_CHANNEL
+        self.name = "Combine RGB"
```

Comment:
Is the name meant to be Combine RGB, or is it supposed to be RGBA? The alpha is optional, I realize, but the name should maybe use RGBA for symmetry with the Separate RGBA node.

Reply:
Good find!